packages/components/time-picker/src/time-picker-com/basic-time-spinner.vue

```diff
@@ -232,12 +232,36 @@ const scrollDown = (step: number) => {
   let now = unref(timePartials)[label]
   const total = currentScrollbar.value === 'hours' ? 24 : 60
   now = (now + step + total) % total
-
+  now = findNextUnDisabled(label, now, step, total)
   modifyDateField(label, now)
   adjustSpinner(label, now)
   nextTick(() => emitSelectRange(label))
 }
-
+const findNextUnDisabled = (
+  type: TimeUnit,
+  value: number,
+  step: number,
+  total: number
+) => {
+  let res = value
+  const list = unref(timeList)[type]
+  let isDisabled: boolean = list[value]
+  if (isDisabled) {
+    const len: number = list.length
+    while (isDisabled) {
+      if (!list[value]) {
+        isDisabled = false
+        res = value
+        break
+      }
+      value = (value + step + total) % total
+      if (value >= len) {
+        value = 0
+      }
```

Comment:
```suggestion
```

This seems redundant. Because of the `%` operator, the `value` is always less than the `total`.